brickflow/engine/task.py

```diff
@@ -97,6 +97,15 @@ class TaskType(Enum):
     NOTEBOOK_TASK = "notebook_task"
 
 
+class TaskRunCondition(Enum):
```

Comment:
Can you please look at this class and may be that can be modified 

```
class BrickflowTriggerRule(Enum):
    ALL_SUCCESS = "all_success"
    NONE_FAILED = "none_failed"

```

Reply:
It was kind of a hack we did at that time, but as it's natively available now - we can deprecate this. 
Probably we can have deprecation message in the next release and ask teams to use `run_if`, In the successive release we can remove it. @stikkireddy What are your thoughts?

Reply:
I can probably reuse and extend this BrickflowTriggerRule feature to pass it as run_if to bundle instead of calculate it during runtime, but it will not be clean in terms of code logic

Reply:
lets deprecate it in another PR after confirming it doesnt mess with any of the airflow adapter logic?

Yea this brickflow trigger rules feature was roughly like 6 months ago when if conditions were still in development.

Can we use the built-in object from the bundles schema instead of managing our own enum instead? @asingamaneni & @grigoriy835

Reply:
nvm on the last part it seems the cli schema treats it as a string and isnt a typed enum. this would be better than letting the user figure out what the values are. 👍

Reply:
@grigoriy835 can you add the import for this enum at the root __init__.py so people can do: from brickflow import TaskRunCondition?

Reply:
i found there is a bunch of code, especially context functions skip_all_except and skip_all_following, that i can't clearly substitute, i think deprecation warning will be better to do not confuse anybody

Reply:
i pushed commit with fixes, added documentation section and fixed linter issues

Reply:
@grigoriy835 @stikkireddy I created this issue for tracking - https://github.com/Nike-Inc/brickflow/issues/56